hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/KeyValueHandler.java

```diff
@@ -460,6 +463,20 @@ ContainerCommandResponseProto handlePutBlock(
       ContainerProtos.BlockData data = request.getPutBlock().getBlockData();
       BlockData blockData = BlockData.getFromProtoBuf(data);
       Preconditions.checkNotNull(blockData);
+      ChunkLayOutVersion layoutVersion =
+          ChunkLayOutVersion.getConfiguredVersion(conf);
+      if (layoutVersion == FILE_PER_BLOCK &&
+          // ChunkManagerDummyImpl doesn't persist to disk, don't check here
+          !chunkManager.getClass()
+              .isAssignableFrom(ChunkManagerDummyImpl.class)) {
```

Comment:
I think layout-specific code would be better added to each `ChunkManager` implementation instead of such `if` and `isAssignableFrom`.  Take a look at `finishWriteChunks()` and `shutdown()`.